UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -167,91 +96,31 @@ private static object TryDeserializeIComparable(JsonReader reader, JsonSerialize
         /// <exception cref="UnitsNetException">Can't serialize 'null' value.</exception>
         public override void WriteJson(JsonWriter writer, object obj, JsonSerializer serializer)
         {
-            Type quantityType = obj.GetType();
+            var quantityType = obj.GetType();
 
             // ValueUnit should be written as usual (but read in a custom way)
             if(quantityType == typeof(ValueUnit))
             {
-                JsonSerializer localSerializer = new JsonSerializer()
+                var localSerializer = new JsonSerializer()
                 {
                     TypeNameHandling = serializer.TypeNameHandling,
                 };
-                JToken t = JToken.FromObject(obj, localSerializer);
+                var t = JToken.FromObject(obj, localSerializer);
 
                 t.WriteTo(writer);
                 return;
             }
 
-            object quantityValue = GetValueOfQuantity(obj, quantityType); // double or decimal value
-            string quantityUnitName = GetUnitFullNameOfQuantity(obj, quantityType); // Example: "MassUnit.Kilogram"
+            var quantity = obj as IQuantity;
 
             serializer.Serialize(writer, new ValueUnit
             {
                 // See ValueUnit about precision loss for quantities using decimal type.
-                Value = Convert.ToDouble(quantityValue),
-                Unit = quantityUnitName
+                Value = quantity.Value,
+                Unit = $"{quantity.QuantityInfo.UnitType.Name}.{quantity.Unit}"
```

Comment:
I would add the comment ` // Example: "MassUnit.Kilogram"` here, it's not entirely obvious what the result is.

Reply:
Done!